doc/superbuild/source/developers/testing/Answers.rst

```diff
@@ -0,0 +1,98 @@
+such as the GitHub Actions virtual machines. 
+
+
+Updating Answer Files
+---------------------
+
+Sometimes it is necessary to update the answer files so that the tests pass, if a change in output
+is expected/desired. Changing output files requires careful verification that the output is still
+"correct". The procedure for updating answer files is as follows:
+
+- For the ``.out`` files embedded in examples, simply overwrite the file with the new output. If the
+  tests still fail, then you will need to download the output file generated on the Jenkins CI
+  machine and commit it. Extracting this file from the Jenkins machine may require help from David
+  or Cody currently.
+
+- For the ``.out`` files in `sundials-codes/answers <https://github.com/sundials-codes/answers>`_:
+
+  #. Navigate to `test/answers` in the sundials repository. If there is nothing in this directory,
+     you need to run `git submodule init && git submodule update`.
+
+  #. Checkout the main branch: ``git checkout main``. 
+
+  #. Create a new branch off of main. It is ideal if you use a branch name that matches the name of
+     the branch that you are developing in sundials on: ``git checkout -b <branch name>``.
+
+  #. Update the relevant ``.out`` files in
+     ``test/answers/linux-ubuntu20.04-x86_64/gcc-9.4.0/<double|single|extended>`` (this is the files
+     used for the GitHub Actions CI). Like with the embedded `.out` files, you can try and use the
+     output generated on your machine, but it may be different enough from what is produced on the
+     GitHub actions to trigger a failure. You can download the output files generated on the GitHub
+     machines by going to `<https://github.com/LLNL/sundials/actions>`_ finding your failing
+     test, clicking it, then at the bottom downloading the "output_files" artifact (you can also find your
+     failing test at the bottom of a PR). The downloaded zip file will be the SUNDIALS build
+     directory. The ``.out`` files you want are in
+     ``build_that_you_downloaded_and_extracted/Testing/output``.
+
+  #. Create a pull-request in `sundials-codes/answers <https://github.com/sundials-codes/answers>`_
+     with your updates. 
+
+
+Adding New Answer Files
+-----------------------
+
+- For the ``.out`` files embedded in examples, simply overwrite the file with the new output. If the
+  tests running on Jenkins still fail, then you will need to download the output file generated on
+  the Jenkins CI machine and commit it. Extracting this file from the Jenkins machine may require
+  help from David or Cody currently.
+
+- For the ``.out`` files in `sundials-codes/answers <https://github.com/sundials-codes/answers>`_:
+
+  #. To add new answers, start by copying the relevant answers in ``linux-ubuntu20.04-x86_64`` to
+     your new directory (following the naming scheme outlined above).
+
+  #. Next Commit these files and open a PR to the "staging" branch.
```

Comment:
Is the staging branch necessary? Recently we've been adding new answer files through a PR to `main`.

Reply:
By new answers, I meant copies of the answers but updated for a different machine. In that case we need to staging branch to be able to diff the out files. Ill clarify in the text.